sdk/servicebus/service-bus/src/serviceBusMessageBatch.ts

```diff
@@ -104,9 +96,7 @@ export class ServiceBusMessageBatchImpl implements ServiceBusMessageBatch {
    * @internal
    * @ignore
    */
-  constructor(context: ClientEntityContext, maxSizeInBytes: number) {
-    this._context = context;
-    this._maxSizeInBytes = maxSizeInBytes;
+  constructor(private _context: ConnectionContext, private _maxSizeInBytes: number) {
```

Comment:
Looking down below we're literally only using this so we can get the .connectionId. This would be an instance where I would use Pick<ConnectionContext, 'connectionId'>.

Reply:
Actually it is used to get the data transformer as well

Reply:
That brings up an excellent point (and not something we have to resolve in this PR). 

But given potential directions we want to pull the encoder - would we probably end up holding onto an instance of the data encoder at the receiver/sender level and not in a connection/entity level?

Reply:
yes, I can log an issue for that.

Reply:
Logged #10620